static/js/chart/draw.ts

```diff
@@ -724,6 +735,80 @@ function drawLineChart(
         dots.on("click", handleDotClick);
       }
     }
+
+    if (highlightOnHover) {
```

Comment:
i see bo's point - this is pretty long and it affects the readability of the overall function.

it would be nice to move it into a helper, even if it does take in a long list of params.  i think passing in both bounding areas for the chart (one for svg transforms, and another for the tooltip / div) could help. and it's fine if it only works for this line chart use case for now.

Reply:
done